src/routes/publish.ts

```diff
@@ -120,62 +118,51 @@ function checkCurrentFileImport(req: AuthedRequest, res: Response): FileImportDT
         req.session.errors = generateViewErrors(undefined, 500, [
             generateError('session', 'errors.session.current_import_missing', {})
         ]);
-        res.redirect(`/${lang}/${req.i18n.t('routes.publish.start', { lng: lang })}/`);
+        res.redirect(`/${lang}/${req.i18n.t('routes.publish.start', { lng: lang })}`);
         return undefined;
     }
     return currentFileImport;
 }
 
-function handleProcessedCSV(processedCSV: UploadDTO | UploadErrDTO, req: AuthedRequest, res: Response) {
-    const lang = req.i18n.language;
-    if (!processedCSV.success) {
-        const err = processedCSV as UploadErrDTO;
-        req.session.errors = generateViewErrors(err.dataset?.id, 500, err.errors);
-        res.redirect(`/${lang}/${req.i18n.t('routes.publish.start', { lng: lang })}/`);
-        return;
-    }
```

Comment:
Rather than wrapping the API return in an object with a success/fail flag, it will now always return the actual DTO we're interested in on success, and throw an exception on fail.